schematics/transforms.py

```diff
@@ -56,11 +56,17 @@ def import_loop(cls, instance_or_dict, field_converter, context=None,
     """
     if not isinstance(instance_or_dict, (cls, dict)):
         raise ModelConversionError('Model conversion requires a model or dict')
-    if mapping is None:
-        mapping = {}
-    data = dict(context) if context is not None else {}
-    errors = {}
 
+    if not env:
+        env = ConfigObject()
+    elif not isinstance(env, ConfigObject):
+        env = ConfigObject(env)
+    env._setdefault('partial', partial)
```

Comment:
Hmm. There may be some misunderstanding here. For the most part, the options are supposed to be specified just like before:

```python
foo.import_data({...}, strict=True, partial=True)
```

This PR does not intend to change that.

An application *may* also pass an `env` object, but there are only two reasons to do so:
  * you want to put some proprietary data in it
  * your custom type or `Model` class wants to change the parameters *in the middle of* the recursive import/export process and propagate the new settings from there on.

What this code does is it sets up the `env` container on the first invocation, and then the options will be frozen thereafter. Letting arguments overwrite the prior state on every call is precisely the problem this PR aims to fix.

Here's an example of how a call stack might look like before:

  1. `Model.__init__({...}, strict=True)`
  2. `Model.convert({...}, strict=True)`
  3. `convert({...}, strict=True)`
  4. `import_loop({...}, strict=True)`
  5. `ModelType.to_native(...)`
  6. `Model.import_data(...)`
  7. `Model.convert(...)`
  8. `convert(...)`
  9. `import_loop(...)`

Okay, back at `import_loop`. What's the value of `strict`? It's `False`, set by `convert()` at step 7. But obviously it should be `True`, since that's what was originally specified.

There were two ways to fix this:

  * make sure every function throughout the chain propagates every possible parameter as an argument, or
  * encapsulate the settings in a single object in the beginning and pass that around instead.

I opted for the latter approach because it's way cleaner and you can now add new options without touching everything.

If there's something that was easy to do before but becomes awkward due to this change, I'm sure we can address it somehow.